src/components/ImageGeneration/GenerationForm/ResourceSelect.tsx

```diff
@@ -52,26 +63,22 @@ function ResourceSelect({
       {!value ? (
         <div>
           <Button
-            variant="default"
+            variant="light"
             leftIcon={<IconPlus size={18} />}
             fullWidth
-            onClick={() =>
-              openResourceSelectModal({
-                title: buttonLabel,
-                onSelect: handleAdd,
-                options: {
-                  ...options,
-                  types: type ? [type] : undefined,
-                },
-              })
-            }
+            onClick={handleOpenResourceSearch}
             {...buttonProps}
           >
             {buttonLabel}
           </Button>
         </div>
       ) : (
-        <ResourceSelectCard resource={value} onUpdate={handleUpdate} onRemove={handleRemove} />
+        <ResourceSelectCard
+          resource={value}
+          onUpdate={handleUpdate}
+          onRemove={allowRemove ? handleRemove : undefined}
+          onSwap={handleOpenResourceSearch}
```

Comment:
I think it would make more sense for this to be a boolean value `showSwap`. It would allow us to keep the functionality internal to the component